src/modules/sv_detection_methods/analyze_sa_tag_method.cpp

```diff
@@ -57,29 +56,82 @@ void retrieve_aligned_segments(std::string const & sa_string, std::vector<Aligne
+        // of the read is lower than the given threshold
+        if (distance_on_read >= -static_cast<int32_t>(max_overlap))
         {
-            Breakend mate1{current.ref_name,
-                           current.orientation == strand::forward ? current.get_reference_end()
-                                                                  : current.get_reference_start(),
-                           current.orientation};
-            Breakend mate2{next.ref_name,
-                           next.orientation == strand::forward ? next.get_reference_start()
-                                                               : next.get_reference_end(),
-                           next.orientation};
-            auto inserted_bases = query_sequence | seqan3::views::slice(current.get_query_end(), next.get_query_start());
-            Junction new_junction{mate1, mate2, inserted_bases, read_name};
-            seqan3::debug_stream << "BND: " << new_junction << "\n";
-            junctions.push_back(std::move(new_junction));
+            int32_t mate1_pos;
+            if (current.orientation == strand::forward)
+            {
+                mate1_pos = current.get_reference_end();
+            }
+            else
+            {
+                mate1_pos = current.get_reference_start();
+            }
+            int32_t mate2_pos;
+            if (next.orientation == strand::forward)
+            {
+                // Correct position of mate 2 for overlapping alignment segments:
+                // Trim alignment of `next` segment at the start to remove overlap
+                if (distance_on_read < 0)
+                {
+                    mate2_pos = next.get_reference_start() - distance_on_read;
+                }
+                else
+                {
+                    mate2_pos = next.get_reference_start();
+                }
+            }
+            else
+            {
+                // Correct position of mate 2 for overlapping alignment segments:
+                // Trim alignment of `next` segment at the end to remove overlap
+                if (distance_on_read < 0)
+                {
+                    mate2_pos = next.get_reference_end() + distance_on_read;
+                }
+                else
+                {
+                    mate2_pos = next.get_reference_end();
+                }
+            }
+            int32_t distance_on_ref = mate2_pos - mate1_pos - 1;
+            // Check that the two consecutive alignment segments either
+            // map to different reference sequences (e.g. translocation, interspersed duplication),
+            // have a large distance on the reference (e.g. deletion, inversion, tandem duplication), or
+            // have a large distance on the read (e.g. insertion)
+            if (current.ref_name != next.ref_name ||
+                abs(distance_on_ref) >= min_length ||
+                distance_on_read >= min_length)
```

Comment:
@SGSSGene: You mentioned the potential problems caused by using unsigned and signed values together. In this line, we again have a comparison of signed with unsigned. Let's say `distance_on_read = -5` and `min_length = 30`. Would `distance_on_read` be converted to unsigned and take a wrong and very large value in this setting? If yes, what can be done about that?
Do we need to cast all unsigned parameters to a signed type every time we use them? If yes, I would rather use signed CLI parameters and add an additional check that their value is non-negative.

Reply:
Yes, this is an issue. If `distance_on_read` is negative it will be promoted to a `uint32_t`. Promotion rules are very weird.
So I think I would also do the same trick as above `distance_on_read >= static_cast<int32_t>(min_length)`
and a assert `assert(min_length <= std::numeric_limits<int32_t>::max())` at the beginning.

And adding a check in the CLI is also a good Idea. You could either use a `signed type`, or just add a check that the uint32v_t value is not larger than `std::numeric_limit<int32_t>::max()`. 


By the way, it might be better to use `std::abs` instead of `abs`. `std::abs` is the c++ version and will auto overload to `int`, `long`, `long long`, `float` etc.... In this particular case it doesn't matter, but `abs` will always call `abs(int)` and might force an implicit cast.